library/CloudBlobHelpers.cs

```diff
@@ -28,26 +28,24 @@ namespace Microsoft.Azure.Toolkit.Replication
     using System.Threading;
     using System.Threading.Tasks;
     using System.Security;
-    using Microsoft.WindowsAzure.Storage;
-    using Microsoft.WindowsAzure.Storage.Table;
-    using Microsoft.WindowsAzure.Storage.Blob;
     using System.Collections.ObjectModel;
+    using global::Azure.Storage.Blobs;
+    using global::Azure.Storage.Blobs.Models;
+    using global::Azure;
+    using global::Azure.Data.Tables;
 
     public class CloudBlobHelpers
     {
-        public static CloudBlockBlob GetBlockBlob(string configurationStorageConnectionString, string configurationLocation)
+        public static BlobClient GetBlockBlob(string configurationStorageConnectionString, string configurationLocation)
         {
-            CloudStorageAccount blobStorageAccount = CloudStorageAccount.Parse(configurationStorageConnectionString);
-            CloudBlobClient blobClient = blobStorageAccount.CreateCloudBlobClient();
-
-            CloudBlobContainer container = blobClient.GetContainerReference(GetContainerName(configurationLocation));
-            if (container.CreateIfNotExists())
+            var blobClient = new BlobServiceClient(configurationStorageConnectionString);
+            var container = blobClient.GetBlobContainerClient(GetContainerName(configurationLocation));
+            if (container.CreateIfNotExists() != null)
             {
-                container.SetPermissions(new BlobContainerPermissions() { PublicAccess = BlobContainerPublicAccessType.Off });
-                container.FetchAttributes(null, new BlobRequestOptions() { ServerTimeout = new TimeSpan(0, 10, 0) });
+                container.SetAccessPolicy(PublicAccessType.None);
```

Comment:
I see that the timeout is removed here, is that not required now ?

Reply:
I didn't see any parameters in new SDK API for BlobRequestOptions to add ServerTimeout. Just remembered that we can do the same with [StorageExtensions.CreateServiceTimeoutScope](https://learn.microsoft.com/en-us/dotnet/api/azure.storage.storageextensions.createservicetimeoutscope?view=azure-dotnet) to set server timeout as 10 minutes with the following change:
```
using (StorageExtensions.CreateServiceTimeoutScope(TimeSpan.FromMinutes(10)))
{
    container.SetAccessPolicy(PublicAccessType.None);
}
```